From 19a2408b36e04a994dddf4ccd31a574186d38689 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Fri, 12 Jul 2024 11:00:33 +0200
Subject: [PATCH 1158/1695] plat-stm32mp1: leverage qsort_int() in PMIC driver

Use recently added qsort_int() helper function.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git b99dacb167392654c5c16701352100347c44a000 plat-stm32mp1: leverage qsort_int() in PMIC driver]
Change-Id: I4a568b5da5fcbb80334c860b3262ed362caa3ad7
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/393543
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/drivers/stm32mp1_stpmic1.c | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_stpmic1.c b/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_stpmic1.c
index c5aed692c..79d3d5297 100644
--- a/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_stpmic1.c
+++ b/core/arch/arm/plat-stm32mp1/drivers/stm32mp1_stpmic1.c
@@ -407,14 +407,6 @@ static TEE_Result pmic_set_voltage(struct regulator *regulator, int level_uv)
 	return TEE_SUCCESS;
 }
 
-static int cmp_int_value(const void *a, const void *b)
-{
-	const int *ia = a;
-	const int *ib = b;
-
-	return CMP_TRILEAN(*ia, *ib);
-}
-
 static size_t refine_levels_array(size_t count, int *levels_uv,
 				  int min_uv, int max_uv)
 {
@@ -422,7 +414,7 @@ static size_t refine_levels_array(size_t count, int *levels_uv,
 	size_t m = 0;
 
 	/* We need to sort the array has STPMIC1 driver does not */
-	qsort(levels_uv, count, sizeof(*levels_uv), cmp_int_value);
+	qsort_int(levels_uv, count);
 
 	/* Remove duplicates and return optimized count */
 	for (n = 1; n < count; n++) {
-- 
2.47.2

